File: src/main/java/robTheSpire/cards/StealACard.java
package robTheSpire.cards;


import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.localization.CardStrings;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import robTheSpire.DefaultMod;
import robTheSpire.actions.StealCardAction;

import robTheSpire.characters.TheDefault;

import static robTheSpire.DefaultMod.makeCardPath;

public class StealACard extends AbstractDefaultCard {
    public static final String ID = DefaultMod.makeID(StealACard.class.getSimpleName());
    private static final CardStrings cardStrings;

    public StealACard() {
        super(ID, cardStrings.NAME, makeCardPath("StealACard.png"), 2, cardStrings.DESCRIPTION, CardType.SKILL, TheDefault.Enums.COLOR_GRAY, CardRarity.BASIC, CardTarget.SELF);
        this.exhaust = true;
    }

    public void use(AbstractPlayer p, AbstractMonster m) {
        this.addToBot(new StealCardAction());
        System.out.println("Using StealACard!");
    }

    public void upgrade() {
        if (!this.upgraded) {
//            this.exhaust = false;
//            ExhaustiveVariable.setBaseValue(this, 2);
//            ExhaustiveField.ExhaustiveFields.isExhaustiveUpgraded.set(this, true);
            this.upgradeBaseCost(1);
            this.upgradeName();
            this.rawDescription = cardStrings.UPGRADE_DESCRIPTION;
            this.initializeDescription();
        }
    }

    @Override
    public AbstractCard makeCopy() {
        return new robTheSpire.cards.StealACard();
    }


    static {
        cardStrings = CardCrawlGame.languagePack.getCardStrings(ID);
        System.out.println("ID for StealACard is: " + ID);
    }
}

File: src/main/java/robTheSpire/cards/StealACard.kt
package robTheSpire.cards

import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.core.CardCrawlGame
import com.megacrit.cardcrawl.localization.CardStrings
import com.megacrit.cardcrawl.monsters.AbstractMonster
import robTheSpire.DefaultMod.Companion.makeCardPath
import robTheSpire.DefaultMod.Companion.makeID
import robTheSpire.actions.StealCardAction
import robTheSpire.characters.TheDefault.Enums.COLOR_GRAY

class StealACard : AbstractDefaultCard(ID, cardStrings!!.NAME, makeCardPath("steal_a_card.png"), 2, cardStrings!!.DESCRIPTION, CardType.SKILL, COLOR_GRAY, CardRarity.BASIC, CardTarget.SELF) {
    override fun use(p: AbstractPlayer, m: AbstractMonster?) {
        addToBot(StealCardAction())
        println("Using StealACard!")
    }

    override fun upgrade() {
        if (!upgraded) {
//            this.exhaust = false;
//            ExhaustiveVariable.setBaseValue(this, 2);
//            ExhaustiveField.ExhaustiveFields.isExhaustiveUpgraded.set(this, true);
            upgradeBaseCost(1)
            upgradeName()
            rawDescription = cardStrings!!.UPGRADE_DESCRIPTION
            initializeDescription()
        }
    }

    override fun makeCopy(): AbstractCard {
        return StealACard()
    }

    companion object {
        val ID = makeID(StealACard::class.java.simpleName)
        private var cardStrings: CardStrings? = null

        init {
            cardStrings = CardCrawlGame.languagePack.getCardStrings(ID)
            println("ID for StealACard is: $ID")
        }
    }

    init {
        exhaust = true
    }
}